pkg/controller/export/export_connection_controller.go

```diff
@@ -0,0 +1,320 @@
+					return e.ObjectNew.GetNamespace() == util.GetNamespace()
+				},
+				DeleteFunc: func(e event.TypedDeleteEvent[*connectionv1alpha1.Connection]) bool {
+					return e.Object.GetNamespace() == util.GetNamespace()
+				},
+				GenericFunc: func(e event.TypedGenericEvent[*connectionv1alpha1.Connection]) bool {
+					return e.Object.GetNamespace() == util.GetNamespace()
+				},
+			},
+		),
+	)
+	if err != nil {
+		return err
+	}
+
+	err = c.Watch(
+		source.Kind(
+			mgr.GetCache(), &statusv1alpha1.ConnectionPodStatus{},
+			handler.TypedEnqueueRequestsFromMapFunc(connectionstatus.PodStatusToConnectionMapper(true)),
+			predicate.TypedFuncs[*statusv1alpha1.ConnectionPodStatus]{
+				CreateFunc: func(e event.TypedCreateEvent[*statusv1alpha1.ConnectionPodStatus]) bool {
+					return e.Object.GetNamespace() == util.GetNamespace()
+				},
+				UpdateFunc: func(e event.TypedUpdateEvent[*statusv1alpha1.ConnectionPodStatus]) bool {
+					return e.ObjectNew.GetNamespace() == util.GetNamespace()
+				},
+				DeleteFunc: func(e event.TypedDeleteEvent[*statusv1alpha1.ConnectionPodStatus]) bool {
+					return e.Object.GetNamespace() == util.GetNamespace()
+				},
+				GenericFunc: func(e event.TypedGenericEvent[*statusv1alpha1.ConnectionPodStatus]) bool {
+					return e.Object.GetNamespace() == util.GetNamespace()
+				},
+			},
+		),
+	)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// +kubebuilder:rbac:groups=connection.gatekeeper.sh,resources=*,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=status.gatekeeper.sh,resources=*,verbs=get;list;watch;create;update;patch;delete
+func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	log.Info("Reconcile request", "namespace", request.Namespace, "name", request.Name)
+
+	if request.Name != r.auditConnectionName {
+		msg := fmt.Sprintf("Ignoring unsupported connection name %s. Connection name should align with flag --audit-connection set or defaulted to %s", request.Name, r.auditConnectionName)
+		log.Info(msg, "namespace", request.Namespace)
+		return reconcile.Result{}, nil
+	}
+
+	deleted := false
+	connObj := &connectionv1alpha1.Connection{}
+	err := r.reader.Get(ctx, request.NamespacedName, connObj)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+		deleted = true
+	}
+
+	if deleted {
+		err := r.system.CloseConnection(request.Name)
+		if err != nil {
+			return reconcile.Result{Requeue: true}, deleteStatus(ctx, r.writer, request.Namespace, request.Name, r.getPod)
+		}
+		log.Info("removed connection", "name", request.Name)
+		return reconcile.Result{}, deleteStatus(ctx, r.writer, request.Namespace, request.Name, r.getPod)
+	}
+
+	err = r.system.UpsertConnection(ctx, connObj.Spec.Config.Value, request.Name, connObj.Spec.Driver)
+	if err != nil {
+		// Reset the active connection status to false if UpsertConnection fails
+		activeConnection := false
```

Comment:
is there a need to change `active` here? this will be overwritten regardless from audit if publish fails for all messages. Thinking out loud.